content/docs/user-guide/remote-storage.md

```diff
@@ -0,0 +1,200 @@
+identical. DVC does a similar trick for directories, ensuring that directory
+changes (e.g. adding new files) are also detectable and trackable.
+
+💡 The specific type of hash used by DVC currently is the
+[MD5 hash](https://en.wikipedia.org/wiki/MD5).
+
+Let's look at a simple example: the file `a.dat`. After adding the file to the
+project with `dvc add a.dat`, a new file `a.dat.dvc` is created in the project
+folder. The new file contains the hash of `a.dat`, which can be looked at by
+examining the file:
+
+```dvc
+$ cat a.dat.dvc
+outs:
+- md5: bba40b7807c80d3f44787b9c6a4aabee
+  size: 1047565
+  path: a.dat
+```
+
+The original `a.dat` has now been renamed to `bb\a40b7807c80d3f44787b9c6a4aabee`
+and moved to the local cache. The version of `a.dat` that we now see in the
+project directory is now a
+[file link](doc/user-guide/large-dataset-optimization#file-link-types-for-the-dvc-cache)
+that refers to the hash-named version of the file.
+
+💡 Note that DVC supports file links that are
+[reflinks](https://blog.ram.rachum.com/post/620335081764077568/symlinks-and-hardlinks-move-over-make-room-for)
+(the default, if possible), hard links, soft links, and basic copies. Please
+check your operating systems capabilities and configure DVC to use the
+appropriate file link type using `dvc config` if needed.
+
+When the project is committed and pushed, the version of `a.dat` in the cache
+will be copied to remote storage. If `a.dat` changes later and is committed
+again, then a new hash for the file will be computed, a new version of the file
+will be created in the cache, and the file link in the project directory is
+updated.
+
+The next brief example shows a directory `myDir` tracked by DVC containing two
+files `a` and `b`:
+
+![Local cache and remote storage](/img/cache_structure.png)
+
+The cache in this example contains three versions of the directory. When the
+project was first committed, the directory contained only the file `a`, as
+illustrated by the directory in the cache with hash starting `6e..`. The second
+version of the directory was committed after the file `b` was added and a new
+hash `22...` for the directory was calculated. The third version of the
+directory was committed after `b` was changed. This is reflected in new hashes
+for both the directory and `b` (`ef...` and `6d...` specifically). DVC is
+therefore storing two versions of `b` in the cache, but only one copy of `a`
+(since, naturally, file `a` has not changed). Also shown in the figure is the
+file `myDir.dvc`, the metadata file specifying the MD5 hash of the
+``current'' version of `myDir` in the cache.
+
+Both remote storage and local cache use the same format for organising and
+naming your project files. If you have just committed and pushed your changes
+from the local workstation to the remote storage, there should be identical
+folders in both places.
+
+## Sharing Files via Remote Storage
```

Comment:
Finally, this section could probably be more hands-on as well, showing push and pull and maybe even another diagram. Take a look for example at [this page which we recently removed](https://github.com/iterative/dvc.org/pull/3275/files#diff-d67b1c8c21df1b27b1051217601d3cb1572e3eb34d0c2fa1d614e0ece7a3ca01) because it wasn't in the right area of our docs (but the content is still relevant and can be recovered to some extent here).

Also, let's make the title more general? And it's nice to mention ML models some times in the context of data management (which includes local and remote storage). So here's a new title suggestion:

```suggestion
## Sharing Data and Models
```